src/main/java/it/pagopa/pn/delivery/svc/NotificationAttachmentService.java

```diff
@@ -225,13 +227,18 @@ public FileInfos computeFileInfo(FileDownloadIdentify fileDownloadIdentify, Inte
         }
 
         log.info("downloadDocumentWithRedirect with fileKey={} name:{} - iun={}", fileKey, name, iun);
-
-        FileDownloadResponse r = this.getFile(fileKey);
-        fileName = buildFilename(iun, name, r.getContentType());
-
-        log.info("downloadDocumentWithRedirect with fileKey={} filename:{} - iun={}", fileKey, fileName, iun);
-
-        return new FileInfos( fileName, r );
+        try {
+            FileDownloadResponse r = this.getFile(fileKey);
+            fileName = buildFilename(iun, name, r.getContentType());
+
+            log.info("downloadDocumentWithRedirect with fileKey={} filename:{} - iun={}", fileKey, fileName, iun);
+            return new FileInfos( fileName, r );
+        } catch (Exception exc) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)